src/components/Header/Header.tsx

```diff
@@ -1,53 +1,113 @@
+  handleFocusEvent: (_e: React.FocusEvent) => {},
+});
 
-const Header = ({ user, onLogin, onLogout, onCreateAccount }: HeaderProps) => (
-  <header>
-    <div className="storybook-header">
-      <div>
-        <svg width="32" height="32" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
-          <g fill="none" fillRule="evenodd">
-            <path d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z" fill="#FFF" />
-            <path d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z" fill="#555AB9" />
-            <path d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z" fill="#91BAF8" />
-          </g>
-        </svg>
-        <h1>Acme</h1>
+const Header = ({
+  logo,
+  logoText,
+  className,
+  children,
+  toggleOpenText = 'Open Menu',
+  toggleCloseText = 'Close Menu',
+}: React.PropsWithChildren<HeaderProps>) => {
+  const [toggleText, setToggleText] = React.useState(toggleOpenText);
+  const [toggleState, setToggleState] = React.useState(false);
+  const [expandedItem, setExpandedItem] = React.useState(defaultMobileMenuLabel);
+  const expanded = expandedItem !== defaultMobileMenuLabel;
+  const handleMenuToggle = function () {
+    console.log('handleMenuToggle');
+    setToggleState((prev) => !prev);
+    setToggleText((prev) => (prev === toggleOpenText ? toggleCloseText : toggleOpenText));
+    setExpandedItem(defaultMobileMenuLabel);
+  };
+  const handleSelection = function (label: string) {
+    console.log('handleSelection', label);
+    setExpandedItem((prev) => (prev === defaultMobileMenuLabel ? label : defaultMobileMenuLabel));
+  };
+  const handleFocusEvent = (e: React.FocusEvent) => {
+    console.log('handleFocusEvent', e.target, e.relatedTarget);
+    console.log(e.target.parentNode !== e.relatedTarget?.parentNode, e.target);
+    if (e.target.parentNode?.parentNode !== e.relatedTarget?.parentNode?.parentNode) {
+      console.log('Something happened', e.target.parentNode, e.relatedTarget?.parentNode);
+      setExpandedItem(defaultMobileMenuLabel);
+    }
+  };
+  return (
+    <header className={classnames(`${px}-header`, className)}>
+      <button
+        type="button"
+        onClick={handleMenuToggle}
+        className={classnames(`${px}-header__toggle-btn`, {
+          [`${px}-header__toggle-btn--open`]: toggleText === toggleCloseText,
+        })}
+      >
+        <span>{toggleText}</span>
+      </button>
+      <h1 className={`${px}-header__logo`} tabIndex={toggleText === toggleOpenText ? 0 : -1}>
+        <img src={logo} height="14" alt={logoText} />
+      </h1>
+      <div className={classnames(`${px}-header__nav`, { [`${px}-header__nav--open`]: toggleState })}>
+        <HeaderContext.Provider
+          value={
+            {
+              expandedItem,
+              setExpandedItem: (item: string) => setExpandedItem(item),
+              expanded,
+              handleSelection: (label: string) => handleSelection(label),
```

Comment:
I think you maybe return the event as well so that consumers might be able to get more information out of the event.  we need to make sure this label doesn't change when the selected value is initialized.  This might need to be `(id: string)`